src/components/MenuItemList.js

```diff
@@ -12,17 +14,38 @@ const defaultProps = {
     menuItems: [],
 };
 
-const MenuItemList = props => (
-    <>
-        {_.map(props.menuItems, menuItemProps => (
-            <MenuItem
-                key={menuItemProps.title}
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...menuItemProps}
-            />
-        ))}
-    </>
-);
+const MenuItemList = (props) => {
+    let popoverAnchor;
+
+    /**
+     * Handle the secondary interaction for a menu item.
+     *
+     * @param {*} link the menu item link or function to get the link
+     * @param {Event} e the interaction event
+     */
+    const secondaryInteraction = (link, e) => {
+        if (typeof link === 'function') {
+            link().then(url => ReportActionContextMenu.showContextMenu(CONTEXT_MENU_TYPES.LINK, e, url, popoverAnchor));
+        } else if (!_.isEmpty(link)) {
+            ReportActionContextMenu.showContextMenu(CONTEXT_MENU_TYPES.LINK, e, link, popoverAnchor);
+        }
+    };
+
+    return (
+        <>
+            {_.map(props.menuItems, menuItemProps => (
+                <MenuItem
+                    key={menuItemProps.title}
+                    onSecondaryInteraction={_.isEmpty(menuItemProps.link) ? e => secondaryInteraction(menuItemProps.link, e) : undefined}
```

Comment:
This works but I don't understand why. 
Why are we calling secondaryInteraction is link is empty? Shouldn't it be the other way around?

Reply:
It shoud be `!_.isEmpty`. Let me double-check.

Reply:
Nevermind, id doesn't work properly, it opens the context menu only if link is a function

Reply:
Yeah, I'll have to explicitly check if it's a function first before the isEmpty. eslint complains about explicitly converting to Boolean.

Reply:
I haven't tested, but maybe `_.isUndefined` will be of help

Reply:
That's a better option, thanks.